### PR TITLE
Stop reading incomplete stripes and skip rows.

### PR DESCRIPTION
**Pull request by kreuzerkrieg**
Changelog category (leave one):
- Bug Fix (user-visible misbehaviour in official stable or prestable release)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix ORC stripe reading

Detailed description / Documentation draft:
ORC stripes were not fully read since `ReadNext` does not ensure the whole stripe read, `ReadAll` looks like descent replacement for `ReadNext`

### CODE REVIEW

File: src/Processors/Formats/Impl/ORCBlockInputFormat.cpp [hunk 1]
```diff
@@ -38,37 +37,28 @@ Chunk ORCBlockInputFormat::generate()
...
 
     ArrowColumnToCHColumn::NameToColumnPtr name_to_column_ptr;
     for (const auto & column_name : column_names)
     {
-        arrow::ArrayVector vec = {batch_result->GetColumnByName(column_name)};
-        std::shared_ptr<arrow::ChunkedArray> arrow_column = std::make_shared<arrow::ChunkedArray>(vec);
+        std::shared_ptr<arrow::ChunkedArray> arrow_column = table->GetColumnByName(column_name);
         name_to_column_ptr[column_name] = arrow_column;
     }
     arrow_column_to_ch_column->arrowColumnsToCHChunk(res, name_to_column_ptr);
```
Comment (Avogar) [inline src/Processors/Formats/Impl/ORCBlockInputFormat.cpp:61]: Now we can use just `arrow_column_to_ch_column->arrowTableToCHChunk(res, table);`

**Comment by kreuzerkrieg** [comment]
@Mergifyio update

File: src/Core/Settings.h [hunk 2]
```diff
@@ -593,6 +593,7 @@ class IColumn;
     M(Bool, input_format_null_as_default, true, "For text input formats initialize null fields with default values if data type of this field is not nullable", 0) \
     M(Bool, input_format_arrow_import_nested, false, "Allow to insert array of structs into Nested table in Arrow input format.", 0) \
     M(Bool, input_format_orc_import_nested, false, "Allow to insert array of structs into Nested table in ORC input format.", 0) \
+    M(Int64, row_batch_size, 100'000, "Batch size when reading ORC stripes.", 0) \
```
Comment (azat) [inline src/Core/Settings.h:596]: Maybe it worth name it like other format related settings? And also add a note about which format supports it?

Comment (kreuzerkrieg) [reply src/Core/Settings.h]: right, my bad. Fixing

**Review by Avogar — APPROVED** [review]

**Pull request merged**